plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -746,7 +746,7 @@ class Pub(
     private fun containsFlutterSdk(workingDir: File): Boolean {
         val pubspec = parsePubspec(workingDir.resolve(PUBSPEC_YAML))
 
-        return (pubspec.dependencies?.get("flutter") as? Pubspec.SdkDependency)?.sdk == "flutter"
+        return pubspec.dependencies.orEmpty().values.any { (it as? Pubspec.SdkDependency)?.sdk == "flutter" }
```

Comment:
While at it, maybe simplify via `filterIsInstance` like

    .filterIsInstance<Pubspec.SdkDependency>().any { it.sdk == "flutter" }

and to make this line shorter, also exit early in case of `null` dependencies like

    val dependencies = parsePubspec(workingDir.resolve(PUBSPEC_YAML)).dependencies ?: return false
    return dependencies.values.filterIsInstance<Pubspec.SdkDependency>().any { it.sdk == "flutter" }

Reply:
Logic-wise, I know too little about Flutter to tell whether the change is meaningful / correct. What actual problem does it solve?

Reply:
The existing code seemed wrong to me, because I noticed that one may depend in multiple ways on the flutter SDK.

Reply:
> because I got a hunch that one may depend in multiple ways on the flutter SDK.

I think this is possible, so your implementation seems correct.

Reply:
Thanks @MarcelBochtler !